src/marqo/s2_inference/s2_inference.py

```diff
@@ -15,39 +16,48 @@
 from marqo.s2_inference.errors import (
     VectoriseError, InvalidModelPropertiesError, ModelLoadError,
     UnknownModelError, ModelNotInCacheError, ModelDownloadError)
+from marqo.s2_inference.inference_cache.inference_cache import InferenceCache
 from marqo.s2_inference.logger import get_logger
 from marqo.s2_inference.model_registry import load_model_properties
 from marqo.s2_inference.models.model_type import ModelType
 from marqo.s2_inference.types import *
 from marqo.tensor_search.configs import EnvVars
 from marqo.tensor_search.enums import AvailableModelsKey
 from marqo.tensor_search.models.private_models import ModelAuth
-from marqo.tensor_search.utils import read_env_vars_and_defaults, generate_batches
+from marqo.tensor_search.utils import read_env_vars_and_defaults, generate_batches, read_env_vars_and_defaults_ints
 
 logger = get_logger(__name__)
 
 # The avaiable has the structure:
 # {"model_cache_key_1":{"model" : model_object, "most_recently_used_time": time, "model_size" : model_size}}
-available_models = dict()
+_available_models = dict()
 # A lock to protect the model loading process
 lock = threading.Lock()
 MODEL_PROPERTIES = load_model_properties()
+_marqo_inference_cache = InferenceCache(cache_size=read_env_vars_and_defaults_ints(EnvVars.MARQO_INFERENCE_CACHE_SIZE),
+                                        cache_type=read_env_vars_and_defaults(EnvVars.MARQO_INFERENCE_CACHE_TYPE))
 
 
-def vectorise(model_name: str, content: Union[str, List[str]], model_properties: dict = None,
+def vectorise(model_name: str, content: Union[str, List[Union[str,Image]], List[Image]],
```

Comment:
Can content be a list of str and image mixed? I.e. a list having two types in it

Reply:
Actually no, I should use `List[str], List[Image]`